### ISSUE TITLE
remove_method fails to remove singleton method on cloned object

### ISSUE THREAD

**Issue by eprothro** [jruby/jruby#4228]
### Environment
- jruby 9.1.5.0 (2.3.1) 2016-09-07 036ce39 Java HotSpot(TM) 64-Bit Server VM 25.72-b15 on 1.8.0_72-b15 +jit [darwin-x86_64]
- Darwin Evans-2015-MacBook-Pro.local 15.6.0 Darwin Kernel Version 15.6.0: Mon Aug 29 20:21:34 PDT 2016; root:xnu-3248.60.11~1/RELEASE_X86_64 x86_64
### Expected Behavior

MRI

``` ruby
class Foo
  def bar
    puts "*** class method"
  end
end

object = Foo.new

object.define_singleton_method "bar" do
  puts "*** singleton method"
end

cloned = object.clone

cloned.singleton_methods
#=> bar

if cloned.singleton_methods.include?(:bar)
  cloned.singleton_class.class_eval do
    remove_method :bar
  end
end

cloned.bar
#=> *** class method
```
### Actual Behavior

``` ruby
class Foo
  def bar
    puts "*** class method"
  end
end

object = Foo.new

object.define_singleton_method "bar" do
  puts "*** singleton method"
end

cloned = object.clone

cloned.singleton_methods
#=> bar

if cloned.singleton_methods.include?(:bar)
  cloned.singleton_class.class_eval do
    remove_method :bar
  end
end
```

```
NameError: method 'bar' not defined in #<Class:0x45752059>
    from org/jruby/RubyModule.java:2882:in `remove_method'
    from (irb):20:in `block in evaluate'
    from org/jruby/RubyModule.java:2832:in `module_eval'
    from (irb):19:in `<eval>'
    from org/jruby/RubyKernel.java:995:in `eval'
    from org/jruby/RubyKernel.java:1296:in `loop'
    from org/jruby/RubyKernel.java:1115:in `catch'
    from org/jruby/RubyKernel.java:1115:in `catch'
    from /Users/eprothro/.rbenv/versions/jruby-9.1.5.0/bin/irb:13:in `<main>'
```

**Comment by headius** [comment]
This appears to be fixed by #4233. I'll add another spec to that PR for this issue.

**Comment by headius** [comment]
Fixed by #4233. Specs in that PR.
